src/toil/job.py

```diff
@@ -208,7 +208,7 @@ def addChildFn(self, fn, *args, **kwargs):
         :rtype: toil.job.FunctionWrappingJob
         """
         if PromisedRequirement.convertPromises(kwargs):
-            return self.addChild(PromisedRequirementFunctionWrappingJob(fn, *args, **kwargs))
+            return self.addChild(PromisedRequirementFunctionWrappingJob(fn, *args, **kwargs).encapsulate())
```

Comment:
Explain in a comment why . encapsulate is necessary. And then cross-reference that explanation in those other places where .encapsulate is called.